tls/s2n_cipher_preferences.c

```diff
@@ -598,3 +609,22 @@ int s2n_connection_set_cipher_preferences(struct s2n_connection *conn, const cha
     return 0;
 }
 
+int s2n_connection_is_valid_for_cipher_preferences(struct s2n_connection *conn, const char *version)
+{
+    notnull_check(conn);
+    notnull_check(version);
+    notnull_check(conn->secure.cipher_suite);
+
+    const struct s2n_cipher_preferences *preferences;
+    GUARD(s2n_find_cipher_pref_from_version(version, &preferences));
+
+    for (int i = 0; i < preferences->count; ++i) {
+        if (0 == strcmp(preferences->suites[i]->name, conn->secure.cipher_suite->name) &&
+            /* make sure we dont use a tls version lower than that configured by the version */
+            s2n_connection_get_actual_protocol_version(conn) >= preferences->suites[i]->minimum_required_tls_version) {
```

Comment:
Same problem: `preferences->suites[i]->minimum_required_tls_version` is not cipher preference restriction. you need to use `preference->minimum_protocol_version` . You can even move it out of the loop:

```c
if (s2n_connection_get_actual_protocol_version(conn) < preference->minimum_protocol_version) {
    return 0;
}
```